pkg/controller/core/workload_controller.go

```diff
@@ -137,32 +137,21 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 {
-		// Finish the workload
-		log.V(3).Info("Workload has Rejected admission checks, Finish with failure")
-		err := workload.UpdateStatus(ctx, r.client, &wl, kueue.WorkloadFinished,
-			metav1.ConditionTrue,
-			"AdmissionChecksRejected",
-			fmt.Sprintf("Admission checks %v are rejected", rejectedChecks),
-			constants.KueueName)
-		return ctrl.Result{}, err
-	}
-
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
 		if updated, err := r.reconcileSyncAdmissionChecks(ctx, &wl, cqName); updated || err != nil {
 			return ctrl.Result{}, err
 		}
 	}
 
-	if workload.SyncAdmittedCondition(&wl) {
+	// If the workload is not admitted update its admitted condition if necessary.
```

Comment:
The code below is very readable, we don't need to repeat what it's doing in the comment.
But it could be useful to explain the why.
Why shouldn't we sync if it's already admitted?

Reply:
done